File: Chauffeur/Deliverables/UnknownDeliverable.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Chauffeur.Deliverables
{
    [DeliverableName("unknown")]
    public sealed class UnknownDeliverable : Deliverable, IProvideDirections
    {
        public UnknownDeliverable(TextReader reader, TextWriter writer)
            : base(reader, writer)
        {
        }

        public override async Task<DeliverableResponse> Run(string command, string[] args)
        {
            await Out.WriteLineFormattedAsync("Unknown command '{0}' entered, check `help` for available commands", string.Join(" ", new[] { command }.Concat(args)));
            return await base.Run(command, args);
        }

        public async Task Directions()
        {
            await Out.WriteLineAsync("Seriously, you're asking for help on the unknown command? Good luck with that");
        }
    }
}

File: src/Chauffeur/Deliverables/UnknownDeliverable.fs
namespace Chauffeur.Deliverables

open Chauffeur
open FSharp.Control.Tasks.V2

[<DeliverableName("unknown")>]
type UnknownDeliverable(reader, writer) =
    inherit Deliverable(reader, writer)

    override __.Run name args =
        task {
            let cmd = args
                      |> Array.append [| name |]
                      |> String.concat " "
            do! writer.WriteLineAsync(sprintf "Unknown command '%s' entered, check `help` for available commands" cmd)
            return DeliverableResponse.Continue
        }

    interface IProvideDirections with
        member __.Directions() =
            task {
                do! writer.WriteLineAsync("Seriously, you're asking for help on the unknown command? Good luck with that")
                return true
            }
